File: 0957-minimum-add-to-make-parentheses-valid/solution.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        const int n = s.size();
        stack<int> st ;
        
        int res=0;
        for(int i = 0 ; i < n ; i++){
            
            if(s[i] == '(')st.push(i);
            else{
                if(st.size() == 0) res++;
                else st.pop();
            }
            
        }
        
        return res + st.size() ;
        
    }
};

File: 0957-minimum-add-to-make-parentheses-valid/solution.java
class Solution {
    public int minAddToMakeValid(String s) {
            Stack<Character> st = new Stack<>();
            char[] c = s.toCharArray();
            int res = 0;
            for (int i = 0; i < c.length; i++) {
                if (c[i] == '(') st.push(c[i]);
                else if (st.isEmpty()) {
                    res++;
                } else {
                    st.pop();
                }
            }

            return res + st.size();
    }
}
